script.user.js

```diff
@@ -182,6 +182,48 @@ function loadEmoteSetEditor() {
 }
 
 
+// Tab completion
+function applyComplete() {
+    const textInput = document.getElementById("gcInput");
+
+    function getEmoteCompletionList(partialName) {
+        const emoteNames = Object.keys(emotes);
+        const lowerPartialName = partialName.toLowerCase();
+        return emoteNames.filter((name) =>
+                                 name.toLowerCase().startsWith(lowerPartialName)
+                                );
+    }
+
+    function onTabPressed(event) {
+        event.preventDefault();
+
+        const inputVal = textInput.value.trim();
+        const currentInputWords = inputVal.split(" ");
+        const currentPartialWord = currentInputWords[currentInputWords.length - 1];
+        const completionList = getEmoteCompletionList(currentPartialWord);
+
+        if (completionList.length === 0){
+            return;
+        };
+
+        if (completionList.length === 1) {
```

Comment:
both sides of this if statement are running the exact same code

Reply:
that is the point, the else statement needs work for cycling through the possible emotes

Reply:
if you plan on merging it in this state, then there's no point of having it there

Reply:
hmm, i was wondering if you had any ideas about how to go about tab-cycling the matching emotes since i'm fairly new to js but i think i could remove that for to have this merged

Reply:
it can be in a different PR or I can implement it myself

Reply:
![ok](https://cdn.7tv.app/emote/6268904f4f54759b7184fa72/2x.webp)